transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -458,22 +468,22 @@ public final ChannelHandlerInvoker invoker() {
         }
 
         @Override
-        public final ChannelOutboundBuffer outboundBuffer() {
+        public ChannelOutboundBuffer outboundBuffer() {
             return outboundBuffer;
         }
 
         @Override
-        public final SocketAddress localAddress() {
+        public SocketAddress localAddress() {
             return localAddress0();
         }
 
         @Override
-        public final SocketAddress remoteAddress() {
+        public SocketAddress remoteAddress() {
```

Comment:
You're right, this is another option. In general there are several options how to achieve the functionality.